### ISSUE TITLE
[$40] Can see the project in the list after leaving it

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#3825]
Follow up from https://github.com/appirio-tech/connect-app/issues/3809#issuecomment-607287584

> https://www.screencast.com/t/W8z7PzHdz
> I can see a small issue though when I (Copilot) leave from the project {3:00}, the application reloads and navigates to the project listing page but I can see the project on the list {3:02}

@vikasrohit this issue is due to the delay in reindexing the project when we leave the project and navigate to the project listing page fast.

Is it ok to keep it like this, or should we add an artificial delay when we are leaving the project?

**Comment by vikasrohit** [comment]
I think it is okay for now as user would see 404 after clicking such projects. Or if we can make that artificial delay similar to Join event, when user clicks on Decline in the same popup, that would also work.

**Comment by maxceem** [comment]
Sum up:
- when user clicks **Leave Project** button

   ![image](https://user-images.githubusercontent.com/146016/79316209-b3bc0400-7f0c-11ea-9bd5-f837cfafb332.png)

   - we should add artificial delay [ES_REINDEX_DELAY](https://github.com/appirio-tech/connect-app/blob/dev/src/config/constants.js#L889) **BEFORE** sending the actual request to the server to give time for the project to be reindexed.

- You may see for reference how we already implement the same delay when user clicks Join Project button: https://github.com/appirio-tech/connect-app/blob/dev/src/projects/actions/projectMember.js#L160-L165

   ![image](https://user-images.githubusercontent.com/146016/79316800-8b80d500-7f0d-11ea-9ded-916542b7a6de.png)

- The loader indicator should be shown immediately as we click the button. So during the delay before the actual request user already see the loading indicator.

- This delay shouldn't be accidentally applied to other cases. For example, it shouldn't be applied if one user removes another user.

   ![image](https://user-images.githubusercontent.com/146016/79316704-5d9b9080-7f0d-11ea-8da5-2e5589d374ba.png)

**Comment by sumitdaga** [comment]
@bug-bash-helper assign

**Comment by sumitdaga** [comment]
@bug-bash-helper https://github.com/appirio-tech/connect-app/pull/3909 is ready for review

**Comment by codejamtc** [comment]
@maxceem I can see an error when `Accepting` an invitation ❌ 
https://monosnap.com/file/dQTLWGACoobAqFk0QgkGOKnB2ZhATs

- The manager sends an invitation to `TopConnCustomer` and accepted by the `TopConnCustomer`. I can see another popup after pressing the join button for the first time, and then we press the join button again 404 error displayed

```
api.topcoder-dev.com/v5/projects/9138/:1 Failed to load resource: the server responded with a status of 403 ()
main.05e97bdb06757b545e47.js:130 Uncaught (in promise) Error: Request failed with status code 403
    at e.exports (main.05e97bdb06757b545e47.js:130)
    at e.exports (main.05e97bdb06757b545e47.js:223)
    at XMLHttpRequest.f.<computed> (main.05e97bdb06757b545e47.js:130)
api.topcoder-dev.com/v5/projects/9138/invites/2136:1 Failed to load resource: the server responded with a status of 404 ()
```

**Comment by maxceem** [comment]
@codejamtc great catch. I think it's a regression issue after fixing another issue. But it's not connected with this thicket. Could you please log it as a separate issue for `2.9` milestone.

**Comment by vikasrohit** [comment]
@maxceem is it the same error which I shared with you from production where customer saw error after accepting invite?

**Comment by maxceem** [comment]
@vikasrohit I think no. I guess this error appears as a regression after one of the fixed during Bug Bash as I couldn't reproduce it before, but I can reproduce it now.

**Comment by vikasrohit** [comment]
ok

**Comment by codejamtc** [comment]
@maxceem https://github.com/appirio-tech/connect-app/issues/3948

**Comment by vikasrohit** [comment]
@codejamtc as #3948 is now passed in dev, can we consider this one as unblocked and get it tested?

**Comment by maxceem** [comment]
Thanks @vikasrohit for noticing it. Yes, @codejamtc, could you please, test this issues now.

**Comment by codejamtc** [comment]
Fixed: 🆗 `Copilot` leaving from a project. Didn't check `Manager` leaving a project because the manager already has access to all the projects.
https://monosnap.com/file/jCzEQHZTGRorF3QtjLPcYQX38rhSJD

**Comment by codejamtc** [comment]
QA pass in `Production`: 🆗 Customer Leaving Project
https://monosnap.com/file/rlHSS9YdxtBx73Tm5yx8tmADEWaeaG

@drasticdpk @lakshmiathreya Please check `Manager` and `Copilot` leaving from a project. Thanks

**Comment by drasticdpk** [comment]
Verified with copilot Role.

**Comment by vikasrohit** [comment]
@drasticdpk @codejamtc added label for prod pass and closing the issue.
